controllers/pagerdutyintegration/handleupdate.go

```diff
@@ -28,20 +33,33 @@ func (r *PagerDutyIntegrationReconciler) handleUpdate(pdclient pd.Client, pdi *p
 		// SERVICE_ID and INTEGRATION_ID
 		configMapName string = config.Name(pdi.Spec.ServicePrefix, cd.Name, config.ConfigMapSuffix)
 	)
-
-	pdData, err := pd.NewData(pdi, cd.Spec.ClusterMetadata.ClusterID, cd.Spec.BaseDomain)
-	if err != nil {
-		return err
-	}
-	err = pdData.ParseClusterConfig(r.Client, cd.ObjectMeta.Namespace, configMapName)
+	cm := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: cd.ObjectMeta.Namespace, Name: configMapName}, cm)
 	if err != nil {
-		return err
+		return nil // requeue and wait for the configmap to be created
```

Comment:
I believe returning an error is what would cause this to requeue right? https://github.com/openshift/pagerduty-operator/pull/229/files#diff-cecb7e9ffd55b2eb077a62f26087f4dc1a582b736b8cefc9771ef67840350efbR238-R240

Reply:
If it returns the error, it causes the entire reconcile loop to return the error, which causes the following test case to fail:

```
--- FAIL: TestReconcilePagerDutyIntegration (0.05s)
    --- FAIL: TestReconcilePagerDutyIntegration/Test_Managed,_No_Finalizer,_Not_Deleting,_PD_Not_Setup (0.00s)
        pagerdutyintegration_controller_test.go:1120:
                Error Trace:    pagerdutyintegration_controller_test.go:1120
                Error:          Expected nil, but got: pagerdutyintegration.pdiReconcileErrors{(*errors.StatusError)(0xc0001a0be0)}
                Test:           TestReconcilePagerDutyIntegration/Test_Managed,_No_Finalizer,_Not_Deleting,_PD_Not_Setup
                Messages:       Unexpected Error with Reconcile (1 of 3)
```

I erred ;) on the side of not changing existing behavior; do we want to do something different for this test case?

Reply:
I guess what I'm worried about is that the reconcile loop won't get requeued if we don't return an error here. Maybe it's not a big deal since the reconcile loop will run any time a clusterdeployment gets added so we shouldn't get into a state where something's not configured and a reconcile loop won't run for "a while" (I'm not sure)?

Reply:
I'm fuzzy on this, to be honest. From what I can understand of [the test code](https://github.com/openshift/pagerduty-operator/blob/master/controllers/pagerdutyintegration/pagerdutyintegration_controller_test.go#L1046-L1068), it looks like pagerduty-operator expects to reconcile multiple times to "finalize" things?

I can change that test case to expect an error if you think that's better, though!

Reply:
I think that's ok then, it is true that the `Reconcile()` function won't get automatically called unless this returns an error, but seems like it's safe to assume that it'll get called a bunch anyway. Sorry for the delay here!